src/lib/options/validate-options.ts

```diff
@@ -0,0 +1,173 @@
+import type { Options } from './index';
+
+const boolOpts = [
+  'makeCumulative',
+  'loop',
+  'showIcons',
+  'showGroups',
+  'mouseControls',
+  'keyboardControls',
+  'autorun',
+  'injectStyles',
+  'highlightBars',
+  'selectBars',
+  'fixedScale',
+] as const;
```

Comment:
it would be nice to add more type-safety here (and similar ones) using [`satisfies`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-9.html#the-satisfies-operator)

```suggestion
] as const satisfies Array<keyof Options>;
```